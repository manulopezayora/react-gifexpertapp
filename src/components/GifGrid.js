import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem'
import Loading from './Loading'

import PropTypes from 'prop-types'

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category)

  return (
    <>
      <h3 className="animate__animated animate__lightSpeedInLeft">
        {category}
      </h3>
      {loading && <Loading />}
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}

export default GifGrid
