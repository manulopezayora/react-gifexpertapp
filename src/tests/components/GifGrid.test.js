import React from 'react'
import '@testing-library/jest-dom'

import { shallow } from 'enzyme'
import GifGrid from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'

// Finjir la llamada al custom hook
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el componente <GifGrid />', () => {
  const category = 'Panda'

  test('debería mostrarse correctamente', () => {
    // Falsear la data para el otro test
    useFetchGifs.mockReturnValue({
      // contenido del state de useFetchGifs
      data: [],
      loading: true,
    })

    // debería mostrarse correctamente
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('debería mostrar ítems cuandose cargan imágenes useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://localhost/cualquier/cosa.png',
        title: 'Cualquier cosa',
      },
    ]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    })

    const wrapper = shallow(<GifGrid category={category} />)
    // expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('Loading').exists()).toBe(false)
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  })
})
