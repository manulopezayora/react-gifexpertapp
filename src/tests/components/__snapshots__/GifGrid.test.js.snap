// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> deber√≠a mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__lightSpeedInLeft"
  >
    Panda
  </h3>
  <Loading />
  <div
    className="card-grid"
  />
</Fragment>
`;
